csharp/src/Apache.Arrow/Schema.Builder.cs

```diff
@@ -57,6 +58,20 @@ public Builder Field(Action<Field.Builder> fieldBuilderAction)
                 return this;
             }
 
+#if NETCOREAPP3_1_OR_GREATER
+            public Builder Struct(Type structure, string timezone = null)
+            {
+                PropertyInfo[] properties = structure.GetProperties(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
```

Comment:
Why `NonPublic`?

Reply:
Wondering if it should expose private fields